fdbbackup/FileDecoder.actor.cpp

```diff
@@ -120,6 +131,56 @@ struct DecodeParams {
 		return !(begin >= endVersionFilter || end <= beginVersionFilter);
 	}
 
+	void updateRangeMap() {
+		for (const auto& prefix : prefixes) {
+			rangeMap.insert(prefixRange(StringRef(prefix)), 1);
+		}
+		rangeMap.coalesce(allKeys);
+	}
+
+	bool matchFilters(MutationRef m) {
+		bool match = false;
+		if (isSingleKeyMutation((MutationRef::Type)m.type)) {
+			auto ranges = rangeMap.intersectingRanges(singleKeyRange(m.param1));
+			for ([[maybe_unused]] const auto r : ranges) {
+				if (r.cvalue() == 1) {
+					match = true;
+					break;
+				}
+			}
+		} else if (m.type == MutationRef::ClearRange) {
+			auto ranges = rangeMap.intersectingRanges(KeyRangeRef(m.param1, m.param2));
+			for ([[maybe_unused]] const auto r : ranges) {
+				if (r.cvalue() == 1) {
+					match = true;
+					break;
+				}
+			}
+		} else {
+			ASSERT(false);
+		}
+
+		for (const auto& prefix : prefixes) {
```

Comment:
Is this loop still needed? Isn't the rangeMap created from prefixes? Or this is more of a validation. If so, consider adding an option to disable it so that we can get the benefit of using the rangeMap.

Reply:
Yes. This is for validation.